### PR TITLE
feat(slack): add block content message support

### PR DESCRIPTION
**Pull request by igpetrov**
feat(slack): add block content message support

closes https://github.com/camunda/team-connectors/issues/584

## CX

Screenshot 2024-01-29 at 16 54 33

Screenshot 2024-01-29 at 16 55 25

### CODE REVIEW

File: connectors/slack/src/main/java/io/camunda/connector/slack/outbound/model/ChatPostMessageData.java [hunk 1]
```diff
@@ -6,37 +6,57 @@
...
+      // Temporary workaround related to camunda/zeebe#9859
+      requestBuilder.text(StringEscapeUtils.unescapeJson(text));
+      // Enables plain text message formatting
+      requestBuilder.linkNames(true);
+    }
+
+    if (blockContent != null) {
+      if (!blockContent.isArray()) {
+        throw new ConnectorException("Block section must be an array");
+      }
```
[the remark this answers is not in the document]
Comment (igpetrov) [reply connectors/slack/src/main/java/io/camunda/connector/slack/outbound/model/ChatPostMessageData.java:55]: Good point. Will update in a moment

File: connectors/slack/element-templates/slack-connector.json [hunk 2]
```diff
@@ -161,6 +161,28 @@
         "equals": "conversations.create"
       }
     },
+    {
+      "id": "message.type",
+      "label": "Message type",
+      "group": "message",
+      "type": "Dropdown",
+      "value": "plain_text",
```
Comment (sbuettner) [inline connectors/slack/element-templates/slack-connector.json:169]: Please use `camelCase` as we usually do.

Comment (sbuettner) [reply connectors/slack/element-templates/slack-connector.json:169]: Saying this as we should migrate this one to the element template generator and we should strive for keeping existing patterns for enums.

File: connectors/slack/src/main/java/io/camunda/connector/slack/outbound/model/ChatPostMessageData.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (igpetrov) [reply connectors/slack/src/main/java/io/camunda/connector/slack/outbound/model/ChatPostMessageData.java:56]: Done

File: connectors/slack/element-templates/slack-connector.json (hunk 2 above)
Comment (igpetrov) [reply connectors/slack/element-templates/slack-connector.json]: Done

Comment (sbuettner) [reply connectors/slack/element-templates/slack-connector.json]: thx!

**Pull request merged**